### PR TITLE
blanket: 0.6.0 -> 0.7.0

### LINKED ISSUE #308146: Update request: blanket 0.6.0 → 0.7.0
- Package name: blanket
- Latest released version: 0.7.0

- Current version on the unstable channel: 0.6.0
- Current version on the stable/release channel: 0.6.0

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@jsierles onny

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by afh**
## Description of changes

* Update to 0.7.0
* Modernize package by
  - moving it to `pkgs/by-name`
  - using `src.hash` rather than `src.sha256`
  - avoiding `with lib;` in `meta` (see #208242 for details)
  - sorting `meta` attributes alphabetically
  
Closes #308146

ℹ️ Unfortunately I'm unable to test this PR mysql as I do not have access to a Linux machine. Any help testing this manually in addition the GitHub checks is much appreciated (friendly nudge to @NovaViper 🙂)

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by JohnRTitor — CHANGES_REQUESTED** [review]
`blueprint-compiler` needs to be added to the native build dependencies.

```log
       > Host machine cpu family: x86_64
       > Host machine cpu: x86_64
       > Program blueprint-compiler found: NO
       >
       > data/resources/meson.build:4:12: ERROR: Program 'blueprint-compiler' not found or not executable
       >
       > A full log can be found at /build/source/build/meson-logs/meson-log.txt
```

**Comment by JohnRTitor** [comment]
Also this version update requires `libadwaita` 1.5.0 which is not yet available on nixpkgs. It is expected to be updated as part of GNOME-46 PR https://github.com/NixOS/nixpkgs/pull/291339 

Please hold off until the PR is merged (or you can target the gnome-46 branch I think?)

**Comment by afh** [comment]
Thanks for testing and providing the helpful feedback, @JohnRTitor. Happy to target the `gnome-46` branch, if @amaxine approves.

File: pkgs/by-name/bl/blanket/package.nix
Comment (JohnRTitor) [inline pkgs/by-name/bl/blanket/package.nix:39]: Should be fine in the nativeBuildInputs, its just a compiler not a runtime dependency.

Comment (afh) [reply pkgs/by-name/bl/blanket/package.nix]: Sure thing 🙂 Apologies if I missed that from your previous comment.

**Comment by NovaViper** [comment]
@afh ran `nixpkgs-review` against NixOS/nixpkgs@c10b1bb0 (which is where the `nixos-unstable-small` branch is at currently on my flake), and this was the result of the build

```
blanket> [49/50] Generating data/resources/blueprints with a custom command
blanket> FAILED: data/resources
blanket> /nix/store/63sam12i6zzcv5z644xpyzrhyji01c7c-blueprint-compiler-0.12.0/bin/blueprint-compiler batch-compile data/resources/. ../data/resources ../data/resources/about.blp ../data/resources/preferences.blp ../data/resources/preset-chooser.blp ../data/resources/preset-dialog.blp ../data/resources/preset-row.blp ../data/resources/sound-context-menu.blp ../data/resources/shortcuts.blp ../data/resources/sound-item.blp ../data/resources/volume-row.blp ../data/resources/window.blp
blanket> error: Namespace Adw does not contain a type called AboutDialog
blanket> at ../data/resources/about.blp line 4 column 5:
blanket>    4 |Adw.AboutDialog about {
blanket>      |    ^
blanket> hint: Did you check your spelling?
blanket> hint: Are your dependencies up to date?
blanket>
blanket> ninja: build stopped: subcommand failed.
error: builder for '/nix/store/y2n2dwyv7ahj3q3wiilvbyhmm7x0z17s-blanket-0.7.0.drv' failed with exit code 1;
       last 10 log lines:
       > FAILED: data/resources
       > /nix/store/63sam12i6zzcv5z644xpyzrhyji01c7c-blueprint-compiler-0.12.0/bin/blueprint-compiler batch-compile data/resources/. ../data/resources ../data/resources/about.blp ../data/resources/preferences.blp ../data/resources/preset-chooser.blp ../data/resources/preset-dialog.blp ../data/resources/preset-row.blp ../data/resources/sound-context-menu.blp ../data/resources/shortcuts.blp ../data/resources/sound-item.blp ../data/resources/volume-row.blp ../data/resources/window.blp
       > error: Namespace Adw does not contain a type called AboutDialog
       > at ../data/resources/about.blp line 4 column 5:
       >    4 |Adw.AboutDialog about {
       >      |    ^
       > hint: Did you check your spelling?
       > hint: Are your dependencies up to date?
       >
       > ninja: build stopped: subcommand failed.
       For full logs, run 'nix log /nix/store/y2n2dwyv7ahj3q3wiilvbyhmm7x0z17s-blanket-0.7.0.drv'.
error: 1 dependencies of derivation '/nix/store/n0l5h6yg3085mxf6lqfbfm6gchyc9mrl-review-shell.drv' failed to build
┏━ 1 Errors:
┃ error: builder for '/nix/store/y2n2dwyv7ahj3q3wiilvbyhmm7x0z17s-blanket-0.…
┣━ Dependency Graph:
┃ ┌─ ⚠ blanket-0.7.0 failed with exit code 1 after ⏱ 2s in buildPhase
┃ ⏸ review-shell
┣━━━ Builds
┗━ ∑ ⏵ 0 │ ✔ 0 │ ⏸ 1 │ ⚠ Exited after 1 build failures at 11:02:06 after 6s

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/308159

1 package failed to build:
blanket

Result of `nixpkgs-review pr 308159` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>blanket</li>
  </ul>
</details>
```

**Comment by Aleksanaa** [comment]
> @amaxine Can this be targeted to gnome-46 branch?

The gnome-46 branch was manually merged into the latest staging-next https://github.com/NixOS/nixpkgs/pull/306080, which should be available in a few days.

**Comment by amaxine** [comment]
I'm only responding because of the direct ping, however since my work on the Gnome 46 update I have [left](https://github.com/NixOS/nixpkgs/pull/307171) in protest. It's unclear to me whether the latest board statement will yet lead to meaningful changes, so I'd request you don't ask further assistance of me. Sorry for that not being clear in the Gnome 46 PR itself.

@JohnRTitor @afh please don't target this to `gnome-46`, that branch was merged into staging already (but not in a way that is marked the github PR as merged). You can probably retarget this to `staging-next`, or just wait until that lands in `master`.

@NovaViper this PR won't build on top of current `master` as it depends on newer versions of most likely libadwaita and gtk. Those are already in `staging-next`.

**Comment by JohnRTitor** [comment]
`staging` and `staging-next` are used to test changes that require many rebuilds and testing, and this PR by itself doesn't meet the criteria for targeting staging-next.
The most preferable branch wou`gnome-46` branch is not maintained anymore.

So, moving on,

**Comment by afh** [comment]
Closing this as something went wrong while trying to retarget `staging-next`. Apologies to everyone getting an unnecessary ping on this ☹️

**Pull request closed without merging**
